src/subcommand/list.rs

```diff
@@ -78,23 +87,29 @@ mod tests {
           OutPoint::from_str("4a5e1e4baab89f3a32518a88c31bc87f618f76673e2cc77ab2127b7afdeda33b:5")
             .unwrap(),
           50 * COIN_VALUE,
+          55 * COIN_VALUE,
           5 * COIN_VALUE,
+          0,
           Rarity::Uncommon,
           "nvtcsezkbth".to_string()
         ),
         (
           OutPoint::from_str("4a5e1e4baab89f3a32518a88c31bc87f618f76673e2cc77ab2127b7afdeda33b:5")
             .unwrap(),
           10,
+          100,
           90,
+          5 * COIN_VALUE,
           Rarity::Common,
           "nvtdijuwxlf".to_string()
         ),
         (
           OutPoint::from_str("4a5e1e4baab89f3a32518a88c31bc87f618f76673e2cc77ab2127b7afdeda33b:5")
             .unwrap(),
           1050000000000000,
+          1150000000000000,
           100000000000000,
+          5 * COIN_VALUE + 90,
           Rarity::Epic,
           "gkjbdrhkfqf".to_string()
         )
```

Comment:
In addition to fixing the existing tests, I think we should have tests that are dedicated to the new fields `end` and `offset`. I think ideally their `assert_eq` would only work on the tested fields, so that it looks something like this:
```rust
assert_eq!(outputs.into_iter().map(|output| output.offset).collect(), vec![0, 10, 30]);
```
And they should have speaking names, like `list_calculates_output_offsets`.

In general the idea is to have very good test coverage, but also to use tests to document the code. For example if someone in the future breaks the offset calculation, it'd be really nice if one of the failing tests was very clearly testing offset calculation. Hope this makes sense.

These tests could also be added in `tests/list.rs`. Not sure what I'd prefer.